pkg/mbedtls/include/riot_config.h

```diff
@@ -0,0 +1,3344 @@
+/**
+ * \file config.h
```

Comment:
I might be mistaking, but this files seems similar to what I did with [`lwipopts.h` in lwIP](https://github.com/RIOT-OS/RIOT/blob/master/pkg/lwip/include/lwipopts.h). Does this file _has_ to be exactly this format or can we adapt it to be more RIOT-y? The fact, that this file is 96% comments, makes me think the latter.

I also don't think that you have to be that verbose on documentation, as the documentation (hopefully) lies with `mbedtls` (which you could just link here).

Reply:
The file is a copy-paste from mbed TLS. For sure I could write a similar RIOT(-y) file, but for further integration of the package, I think it simplifies the development process to have a (almost) copy available. With that, a developer (i) sees immediately all available configuration options (&doc) and (ii) can easily look at diffs to the original file during version bump or the like...

Reply:
Mh from the other point of view, I might be interested as a user, which configuration options of `mbedtls` are actually used. Scrolling though this 3300 line file to filter out the 142 lines that are actually important seems cumbersome, especially given, that you want to exclude this file from out `doxygen` build. Furthermore, we might want to make options configurable from the app (like we do e.g. with lwIP...),  which seems a bit more complicated when you need to comment out / comment in defines.

I also don't see the maintenance advantage you propose. If you want to see which options are available in `mbedtls`: go to the `mbedtls` documentation. If there is a version bump and config options change this will usually lead to failing compilation, giving you exactly the names of the macros that are new (and thus missing).

Reply:
From https://github.com/RIOT-OS/RIOT/pull/15671#issuecomment-747636247

> Lets give it some more time and if no one opposes, we'll go with your proposal.

Yes, please @RIOT-OS/maintainers, can we have more thoughts and comments on this?

Reply:
(I haven't checked the code, just followed the discussion)
If we can avoid the 3300 lines of configuration code, that would be better. If it's added, there's no reason to skip it from doxygen, or any other checks.
IIUC, the goal of this file is to provide the configuration of the package. Why not link it to the module mechanism of RIOT to define flags (like `MBEDTLS_SHA256_ALT`) or via a Kconfig file or both (depending on the cases) ?

- for the module version:
```c
ifdef MODULE_MBEDTLS_CIPHER_MODE_CBC
#define MBEDTLS_CIPHER_MODE_CBC
#endif
```

- for the Kconfig version (assuming there's a corresponding Kconfig boolean option):
```c
#if IS_ACTIVE(CONFIG_MBEDTLS_ECP_DP_SECP192R1_ENABLED)
#define MBEDTLS_ECP_DP_SECP192R1_ENABLED
#endif
```